src/osdc/Objecter.cc

```diff
@@ -3095,10 +3098,18 @@ void Objecter::handle_osd_op_reply(MOSDOpReply *m)
     // set rval before running handlers so that handlers
     // can change it if e.g. decoding fails
     if (*pr)
+#if defined(__linux__)
```

Comment:
I'll reverse this logic. To specifically call the translation logic on solaris (and aix) and have the else block the default case so it can handle darwin/freebsd without modifications